### PR TITLE
LVM-mode improvements: drop -namespacesize, wait for device creation, improve locking

### PR DESCRIPTION
**Pull request by okartau**
- Maintaining multiple fixed-size namespaces does not add value, so lets create one namespace per NS-mode.
- add short wait for device appearing as it may not happen instantly and we hit error "stat: no such device"
- improve use of mutex yo avoid keeping lock over whole storage for longer times (i.e. runtime of `shred`)

Resolves: #275
Resolves: #289
Resolves: #280

### CODE REVIEW

**Comment by okartau** [comment]
I add 3rd commit here, also related to LVM mode improvement so I moved it over from #290 (which gets closed).
This is proposed work-around for missing device issues mentioned in #280 and #289. I tried it out (not exact version pushed now, but earlier incarnation where wait() was implemented inside FlushDevice, but the idea remains), and it resolved the issue: I saw cases during 2-volume reference deployment where device path was not there immediately after lvcreate, but was there 100 ms later.

**Comment by okartau** [comment]
This improves also repeated e2e test success rate compared to what #280 initially reported.
I remove WIP from title, please check and comment.

**Comment by avalluri** [comment]
@okartau Can you please squash first two commits(code change + documentation change).

File: pkg/pmem-device-manager/pmd-lvm.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ func (lvm *pmemLvm) CreateDevice(name string, size uint64, nsmode string) error
 				if err != nil {
 					return err
 				}
+				err = WaitDeviceAppears(device)
```
Comment (avalluri) [inline pkg/pmem-device-manager/pmd-lvm.go]: Is this change really required?. If I understood correctly, the issue wasn't with missing /dev mounting in LVM mode?.

**Comment by okartau** [comment]
> Can you please squash first two commits

done, pushed

File: pkg/pmem-device-manager/pmd-lvm.go (hunk 1 above)
Comment (okartau) [reply pkg/pmem-device-manager/pmd-lvm.go]: I think it is needed. The "wait for device" is different issue from "missing /dev mount". My observations have shown that creating of LVolume and following appearance of it's entry in /dev/ is not synchronous operation:
after lvcreate returns and driver code goes on accessing /dev/ entry,
the new device entry either exists in /dev, or appears shortly later.
As it may appear asynchronously after short delay, it makes sense to retry few times instead of failing on 1st try.
With failing we destroy chance to create volume which really was just showing up with a delay.
The combination of lvcreate() + WaitDeviceAPpears() is kind an attempt to make the creation operation synchronous by semantics.

Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go]: @okartau Thanks for the clarification.

**Comment by avalluri** [comment]
Looks good to me
